cpp/src/arrow/dataset/file_ipc.cc

```diff
@@ -173,6 +173,20 @@ Result<ScanTaskIterator> IpcFileFormat::ScanFile(
   return IpcScanTaskIterator::Make(options, fragment);
 }
 
+Future<util::optional<int64_t>> IpcFileFormat::CountRows(
+    const std::shared_ptr<FileFragment>& file, compute::Expression predicate,
+    std::shared_ptr<ScanOptions> options) {
+  if (FieldsInExpression(predicate).size() > 0) {
```

Comment:
Nit: let's add a `ExpressionHasNoFieldRefs()` or `FieldsInExpressionCount()` function so we're not constructing a vector of strings just to check it for emptiness